src/shared/conf-parser.c

```diff
@@ -679,6 +678,45 @@ int config_parse_many(
         return 0;
 }
 
+int config_parse_standard_file_with_dropins_full(
+                const char *root,
+                const char *main_file,    /* A path like "systemd/frobnicator.conf" */
+                const char *sections,
+                ConfigItemLookup lookup,
+                const void *table,
+                ConfigParseFlags flags,
+                void *userdata,
+                Hashmap **ret_stats_by_path,
+                char ***ret_dropin_files) {
+
+        char **conf_paths = CONF_PATHS_STRV("");
+        _cleanup_strv_free_ char **configs = NULL;
+        int r;
+
+        /* Build the list of main config files */
+        r = strv_extend_strv_prepend_append(&configs, root, conf_paths, main_file);
```

Comment:
leftover? (`strv_extend_strv_biconcat`)

Reply:
Also, is it really necessary to prepend `root` here? It's simpler to just use `CHASE_PREFIX_ROOT`?

Reply:
It was a misplaced chunk.

`CHASE_PREFIX_ROOT` would be nicer, but we pass this list to `config_parse_many`, which is called from many places and they all expect to pass a prefixed list. So it'd require a major rework.